ext/bcmath/libbcmath/src/round.c

```diff
@@ -0,0 +1,164 @@
+/* round.c: bcmath library file. */
+/*
+    Copyright (C) 1991, 1992, 1993, 1994, 1997 Free Software Foundation, Inc.
+    Copyright (C) 2000 Philip A. Nelson
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.  (LICENSE)
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to:
+
+      The Free Software Foundation, Inc.
+      59 Temple Place, Suite 330
+      Boston, MA 02111-1307 USA.
+
+    You may contact the author by:
+       e-mail:  philnelson@acm.org
+      us-mail:  Philip A. Nelson
+                Computer Science Department, 9062
+                Western Washington University
+                Bellingham, WA 98226-9062
+
+*************************************************************************/
+
+#include "bcmath.h"
+#include "private.h"
+#include <stddef.h>
+
+void bc_round(bc_num num, zend_long precision, zend_long mode, bc_num *result)
+{
+	/* clear result */
+	bc_free_num(result);
+
+	/*
+	* The following three cases result in an early return:
+	*
+	* 1. When rounding to an integer part, when trying to round to a digit
+	*    larger than the num.
+	* 2. If try to round to the finest digit of num or finer than that,
```

Comment:
What do you mean by "finest" here?

Reply:
I mean "smallest digit". For example, if num is `1.23456`, that digit means `6`.
What is the appropriate way to express it in English?